proposals/data-transfer-proposal.md

```diff
@@ -0,0 +1,99 @@
+# Project Pitch: Data Transfer At Lightning Speed
+*Insuring our data transfer protocols can transfer data from known peers as quickly as possible, at speeds as fast or faster than HTTP*
+
+Authors: @hannahhoward
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+
+Every application that touches the DWeb stack has to transfer data. Traditionally, we’ve had trouble reaching performance in data transfer that compares to traditional web transfer speed. Previous efforts to improve Bitswap have improved but not resolved data transfer performance issues. Filecoin is using Graphsync with promising but mixed results.
+
+Over the past few months, we’ve been refactoring and improving go-graphsync, and building out a strong testbed and metrics to measure performance ([github.com/protocol/beyond-bitswap](https://github.com/protocol/beyond-bitswap))
+
+Between [a strong benchmarking suite](https://github.com/protocol/beyond-bitswap/tree/master/testbed/testbed), [good ideas for using Bitswap and Graphsync in complementary ways](https://github.com/protocol/beyond-bitswap/issues/25), and promising initial measurements ([Graphsync vs Bitswap](https://observablehq.com/@acruikshank/graphsync-vs-bitswap-file-transfer-rates), [Graphsync vs HTTP](https://observablehq.com/@acruikshank/graphsync-vs-http-file-transfer-rates))
+
+#### Assumptions &amp; hypotheses
+
+*   Data transfer speeds matter to all users
+*   Data transfer speeds are a pain point, even absent problems with content routing
```

Comment:
This is my hangup right now - it’s not clear to me this is a major pain point for new and current users. Can we run a very lightweight survey to get some prioritization info?

Reply:
The majority of NFT’s we see are small, too small to trip over this.

There was a complaint about large uploads being slow and sometimes failing but I haven’t seen it tracked down enough to know where the issue is and if this is related to it.